### PR TITLE
Implementation of CAIP-2 for EOSIO blockchains

### PR DESCRIPTION
**Pull request by sebastianmontero**
Initial draft of the CAIP-2 for EOSIO blockchains.

### CODE REVIEW

**Review by ligi — CHANGES_REQUESTED** [review]
Please also add your CAIP to the list in the README

**Review by ligi — APPROVED** [review]
LGTM

**Comment by ligi** [comment]
LGTM - but would love some more eye-ballz on it before the merge  @pedrouid @webmaster128 @rekmarks

File: CAIPs/caip-7.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+## Motivation
+
+See CAIP-2.
+
+## Specification
+
+### EOSIO Namespace
+
+The namespace "EOSIO" refers to the EOSIO open-source blockchain platform.
```
Comment (webmaster128) [inline CAIPs/caip-7.md:30]:
```suggestion
The namespace "eosio" refers to the EOSIO open-source blockchain platform.
```

File: CAIPs/caip-7.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+#### Reference Definition
+
+The definition is delegated to the [EOSIO Transactions Protocol Documentation](https://developers.eos.io/welcome/v2.0/protocol/transactions_protocol/#32-sign-transaction), the [Chain API Plugin Documentation of the EOSIO Developers Manual](https://developers.eos.io/manuals/eos/latest/nodeos/plugins/chain_api_plugin/api-reference/index?query=chain%20id&page=1#operation/get_info) and the pull request that implemented its generation [Chain ID generation implementation pull request](https://github.com/EOSIO/eos/pull/3425). 
+The Chain ID is the SHA256 hash of the genesis state of the chain, and its represented as hexadecimal number of 64 digits
```
Comment (webmaster128) [inline CAIPs/caip-7.md:35]:
```suggestion
The Chain ID is the SHA256 hash of the genesis state of the chain, represented as lower case hexadecimal number of 64 digits.
```

**Review by webmaster128 — CHANGES_REQUESTED** [review]
Oh, and this conflicts with https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md#syntax. The reference part must not exceed 47 characters, which is why we use prefixes in https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-4.md

**Comment by webmaster128** [comment]
The alternative to prefixes is to use [a different encoding](https://en.wikipedia.org/wiki/Binary-to-text_encoding#Encoding_standards) for the 32 bytes. CAIP-2 is case-sensitive and you need 32/47 = 68 % efficiency. So candidates are [Base58](https://de.wikipedia.org/wiki/Base58) or [Base62](https://en.wikipedia.org/wiki/Base62) (no idea how this encoding works but promising in theory).

However, I think it is easier for human consumption to use a prefix of the default encoding than to re-encode it.

**Comment by sebastianmontero** [comment]
I've updated the reference definition, to use a 32 character prefix of the Chain ID

File: CAIPs/caip-7.md [hunk 3]
```diff
@@ -32,7 +32,7 @@ The namespace "eosio" refers to the EOSIO open-source blockchain platform.
 #### Reference Definition
 
 The definition is delegated to the [EOSIO Transactions Protocol Documentation](https://developers.eos.io/welcome/v2.0/protocol/transactions_protocol/#32-sign-transaction), the [Chain API Plugin Documentation of the EOSIO Developers Manual](https://developers.eos.io/manuals/eos/latest/nodeos/plugins/chain_api_plugin/api-reference/index?query=chain%20id&page=1#operation/get_info) and the pull request that implemented its generation [Chain ID generation implementation pull request](https://github.com/EOSIO/eos/pull/3425). 
-The Chain ID is the SHA256 hash of the genesis state of the chain, represented as lower case hexadecimal number of 64 digits.
+The Chain ID, as defined by EOSIO, is the SHA256 hash of the genesis state of the chain, represented as lower case hexadecimal number of 64 digits. In order to fit the reference restrictions, the format is a 32 character prefix of the Chain ID.
```
Comment (webmaster128) [inline CAIPs/caip-7.md:35]:
```suggestion
The Chain ID, as defined by EOSIO, is the SHA256 hash of the genesis state of the chain, represented as lower case hexadecimal number of 64 digits. In order to fit the CAIP-2 reference format, a 32 character prefix of the Chain ID is used.
```

**Review by webmaster128 — COMMENTED** [review]
Some wording 💅 but otherwise 👍

**Review by webmaster128 — APPROVED** [review]
Cool and welcome

**Comment by sebastianmontero** [comment]
Great! Thanks!

**Comment by webmaster128** [comment]
@ligi can we merge this? I don't have permissions to do so.

**Pull request merged**